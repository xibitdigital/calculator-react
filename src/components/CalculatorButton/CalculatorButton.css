.calculator-button {
    background-color: #aaa;
    border: 1px solid #ccc;
    outline: 0;
    box-sizing: border-box;
    font-size: 21px;
    font-weight: 300;
    height: 47px;
    width: 57px;
}

.calculator-button:active {
    background-color: #eee !important;
}

.calculator-button--wide {
    flex-grow: 1;
}
