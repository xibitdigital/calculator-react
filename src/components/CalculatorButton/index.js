import React from 'react'
import PropTypes from 'prop-types'
import './CalculatorButton.css'

export default function CalculatorButton({
    backgroundColor,
    callback,
    color,
    value,
    wide,
}) {
    return (
        <button
            onClick={callback}
            className={`calculator-button${
                wide ? ' calculator-button--wide' : ''
            }`}
            style={{ backgroundColor, color }}
        >
            {value}
        </button>
    )
}

CalculatorButton.propTypes = {
    backgroundColor: PropTypes.string,
    callback: PropTypes.func,
    color: PropTypes.string,
    wide: PropTypes.bool,
}

CalculatorButton.defaultProps = {
    backgroundColor: '#e0e0e0',
    color: 'black',
    wide: false,
}
