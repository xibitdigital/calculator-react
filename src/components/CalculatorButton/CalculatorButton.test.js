import React from 'react'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'

import sinon from 'sinon'

import CalculatorButton from '.'

test('Display a CalculatorButtonu ', () => {
    const operator = jest.fn()
    const calculatorButton = shallow(
        <CalculatorButton
            backgroundColor="#aaa"
            callback={operator}
            value="Foo"
        />
    )
    expect(shallowToJson(calculatorButton)).toMatchSnapshot()
})

test('CalculatorButton calls props.callback on click', () => {
    const spy = sinon.spy()
    const calculatorButton = shallow(<CalculatorButton callback={spy} />)

    calculatorButton.find('.calculator-button').simulate('click')

    sinon.assert.calledOnce(spy)
})
