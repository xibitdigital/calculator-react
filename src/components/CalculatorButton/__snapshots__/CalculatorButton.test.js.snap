// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display a CalculatorButtonu  1`] = `
<button
  className="calculator-button"
  onClick={[Function]}
  style={
    Object {
      "backgroundColor": "#aaa",
      "color": "black",
    }
  }
>
  Foo
</button>
`;
